### ISSUE TITLE
Tree: set aria-setsize and aria-posinset

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#27900]
The tree/list is virtualized and as such a screen reader cannot make a correct statement on the number of entries. We should leverage [aria-setsize](https://www.w3.org/TR/wai-aria/states_and_properties#aria-setsize) and [aria-posinset](https://www.w3.org/TR/wai-aria/states_and_properties#aria-posinset) for that purpose within the tree and list widgets. The widgets know the number of elements to render and the currently selected items position in the set.

/cc @joaomoreno @isidorn

**Comment by isidorn** [comment]
Makes sense, I can look into it in june

**Comment by isidorn** [comment]
Accidently closed this by PR which only address the list widget, still gotta do it for the tree

**Comment by isidorn** [comment]
After looking into this with @joaomoreno we have fixed this issue for the [list](https://github.com/Microsoft/vscode/commit/edcbf6525aa19f495bc4d9b4e9edb6dcb12bd3ee), however for the tree the situation is a bit more complex:
* It is not exactly specified what the `aria-posinset` and `aria-setsize` should be with respect to the tree structure. Should they be absolute or only relative to the siblings.
* In both of the cases we have an issue that the tree smartly chooses the minimal set of elements that need to be re-rendered, so if an element A gets removed and element B is above it, B will not get re-rendered and will still have the wrong `aria-setsize` attribute.

Due to that we would need a bigger refactoring in order to fix this for the tree. So renaming this issue and pushing it to backlog for now.

**Comment by alexandrudima** [comment]
The originating issue -- https://github.com/Microsoft/monaco-editor/issues/444 -- came in w.r.t. the Command Palette that displays a list. @bpasero perhaps the root issue is that the implementation of the Command Palette uses a tree to display a list and that makes it difficult to express the list to Screen Readers ?

**Comment by isidorn** [comment]
+1 for making the command palette use the list. I could also look into this when I get cycles if @bpasero does not mind.

**Comment by bpasero** [comment]
@isidorn yeah +1 for using list for quick open, that would be very nice 👍

**Comment by isidorn** [comment]
I have created this issue to track it https://github.com/Microsoft/vscode/issues/29096
